### PR TITLE
Workflows

### PR DESCRIPTION
**Pull request by baconpaul**
DO NOT MERGE

### CODE REVIEW

File: .github/workflows/pullreq.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+      - name: Build project
+        run: |
+          cmake -S . -B ./build ${{ matrix.cmakeargs }} -DCMAKE_BUILD_TYPE=Debug -DCLAP_HELPERS_BUILD_TESTS=TRUE
+          cmake --build ./build --config Debug clap-helpers-tests
+
+      - name: Run Tests
+        shell: bash
+        run: |
+          todo run tests
```
Comment (Trinitou) [inline .github/workflows/pullreq.yml]: Just as a reminder: Here we still need a proper command to run the tests

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,34 +1,63 @@
...
+            SOURCE_DIR cpm/clap
+    )
+
+    add_subdirectory(${CMAKE_BINARY_DIR}/cpm/clap)
+endif()
+target_link_libraries(${PROJECT_NAME} INTERFACE clap)
+
+if (${CLAP_HELPERS_BUILD_TESTS})
+    if (NOT DEFINED ${CLAP_HELPERS_TESTS_CXX_STANDARD})
+        message(STATUS "${PROJECT_NAME}: defaualting to c++11")
```
Comment (Trinitou) [inline CMakeLists.txt]:
```suggestion
        message(STATUS "${PROJECT_NAME}: defaulting to c++11")
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,34 +1,63 @@
...
+add_library(${PROJECT_NAME} INTERFACE)
+target_include_directories(${PROJECT_NAME} INTERFACE include)
 
-    if (Catch2_FOUND)
-        add_executable(clap-helpers-tests EXCLUDE_FROM_ALL
+if (NOT TARGET clap)
+    message(STATUS "${PROJECT_NAME}: Downloading CLAP using CPM")
+
+    message(STATUS "${PROJECT_NAME}: including CPM")
+    include(cmake/external/CPM.cmake)
```
Comment (Trinitou) [inline CMakeLists.txt:15]: I thought about this a bit: I think some CLAP_HELPERS_DOWNLOAD_DEPENDENCIES option (like for the wrapper) would be good here as well

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,34 +1,63 @@
...
+            DOWNLOAD_ONLY TRUE
+            SOURCE_DIR cpm/clap
+    )
+
+    add_subdirectory(${CMAKE_BINARY_DIR}/cpm/clap)
+endif()
+target_link_libraries(${PROJECT_NAME} INTERFACE clap)
+
+if (${CLAP_HELPERS_BUILD_TESTS})
+    if (NOT DEFINED ${CLAP_HELPERS_TESTS_CXX_STANDARD})
```
Comment (Trinitou) [inline CMakeLists.txt]: I'm not completely sure about DEFINED syntax but in clap-wrapper the equivalent code is written like this:
```suggestion
    if (NOT DEFINED CLAP_HELPERS_TESTS_CXX_STANDARD)
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,34 +1,63 @@
...
-        add_executable(clap-helpers-tests EXCLUDE_FROM_ALL
+if (NOT TARGET clap)
+    message(STATUS "${PROJECT_NAME}: Downloading CLAP using CPM")
+
+    message(STATUS "${PROJECT_NAME}: including CPM")
+    include(cmake/external/CPM.cmake)
+    CPMAddPackage(
+            NAME "clap"
+            GITHUB_REPOSITORY "free-audio/clap"
+            GIT_TAG "next"
```
Comment (Trinitou) [inline CMakeLists.txt:19]: this will be set to `"1.2.0"` once that one is ready, right?

**Comment by abique** [comment]
@baconpaul I've converted the PR to draft as you wrote do not merge.

File: CMakeLists.txt (hunk 5 above)
Comment (baconpaul) [reply CMakeLists.txt:19]: yup

File: CMakeLists.txt (hunk 3 above)
Comment (baconpaul) [reply CMakeLists.txt:15]: yeah ok i added a force download option

File: .github/workflows/pullreq.yml (hunk 1 above)
Comment (baconpaul) [reply .github/workflows/pullreq.yml]: yes just need pipelines running to test this so will leave it here for now as a forced fail

**Pull request closed without merging**